cloudevents/abstract/event.py

```diff
@@ -0,0 +1,97 @@
+import abc
+import typing
+from typing import TypeVar
+
+
+class CloudEvent(abc.ABC):
+    """
+    Used internally for multi-framework event type integration
+    """
+
+    @classmethod
+    @abc.abstractmethod
+    def create(
+        cls,
+        attributes: typing.Dict[str, typing.Any],
+        data: typing.Optional[typing.Any],
+    ) -> "CloudEvent":
+        """
+        Factory function.
+        We SHOULD NOT use the __init__ function as the factory because
+        we MUST NOT assume how the __init__ function looks across different python
+        frameworks
+        """
+        pass
```

Comment:
Why should we have this method? I don't think it worth having it at all. this forces us to implement it everywhere (and any other potential implementation to have it) and I don't see any benefits in having it.

IMO, the whole idea of the "abstract" implementation is just to provide the contract of what should be 100% available.

Reply:
OK, so you're using it in the http/json methods.

But It should return a generic `AnyCloudEvent` then. Cause implementations are going to return their own instances and thus types. This probably should be handled by generics.

Reply:
yes, good idea